GCDWebServer/Core/GCDWebServer.m

```diff
@@ -592,6 +594,11 @@ - (BOOL)_start:(NSError**)error {
       CFStreamError streamError = {0};
       CFNetServiceRegisterWithOptions(_registrationService, 0, &streamError);
 
+      if (txtRecordDictionary) {
```

Comment:
This code should be before the service is registered I think so moved to line 590